ui/src/app/shared/pod-name.ts

```diff
@@ -1,8 +1,24 @@
+import {NodeStatus} from '../../models';
+
+export const POD_NAME_V1 = 'v1';
+export const POD_NAME_V2 = 'v2';
+
 export const maxK8sResourceNameLength = 253;
 export const k8sNamingHashLength = 10;
 
 // getPodName returns a deterministic pod name
-export const getPodName = (workflowName: string, nodeName: string, templateName: string, nodeID: string): string => {
+export const getPodName = (workflowName: string, nodeName: string, templateName: string, nodeID: string, version: string): string => {
+    if (version === POD_NAME_V2) {
+        if (workflowName === nodeName) {
+            return workflowName;
+        }
+
+        const prefix = ensurePodNamePrefixLength(`${workflowName}-${templateName}`);
+
+        const hash = createFNVHash(nodeName);
+        return `${prefix}-${hash}`;
+    }
+
```

Comment:
huh this looks like the logic in `workflow/util/pod_name.go` _before_ this PR. this PR changed that file a bit to invert the conditionals, but it's also slightly semantically different.

noticed this was divergent in https://github.com/argoproj/argo-workflows/pull/12928#issuecomment-2068605536, working on fixing this

Reply:
Resolved in #12964